subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/normalization/KotlinApiClassExtractor.kt

```diff
@@ -89,7 +89,8 @@ class KotlinApiMemberWriter(apiMemberAdapter: ClassVisitor, val inlineMethodWrit
     override fun writeMethod(method: MethodMember) {
         when {
             method.isInternal() -> return
-            method.isInline() -> inlineMethodWriter.writeMethod(method)
+            // TODO: detect lambdas in public inline methods and treat them as ABI, then inlineMethodWriter.writeMethod(method)
+            method.isInline() -> throw GradleException("Can't use compile avoidance with inline functions")
```

Comment:
Should we log a WARN here to let the users know when we can not use compile avoidance?

Reply:
When that happens, what's the scope of the problem? I mean, does it only affect the ability of one JAR/directory to be used in compilation avoidance mode?

Reply:
This path is reached when computing a fingerprint of a jar that is on the classpath of a buildscript.
If that jar happens to have a public inline method, then instead of computing a fingerprint of its ABI, the full fingerprint for that jar would be computed. That is - this affects the ability of that one jar to be used in compilation avoidance.
I then think this is not an issue if this jar happens to be an external library on the classpath (less likely to change, fine if buildscripts are recompiled on version bump). In which case we might only want to warn about this if we can detect that we are computing an ABI of a locally built jar (more likely to change) - that is either buildSrc or an included build.

We should also document this.

Reply:
> I then think this is not an issue if this jar happens to be an external library on the classpath (less likely to change, fine if buildscripts are recompiled on version bump). In which case we might only want to warn about this if we can detect that we are computing an ABI of a locally built jar (more likely to change) - that is either buildSrc or an included build.

👍